File: C/rng.c
#include <stdio.h>
#include <stdlib.h>

long long seed=0x5DEECE66DLL;

void setSeed (long long newSeed) {
  seed=(newSeed ^ 0x5DEECE66DLL) & ((1LL << 48) - 1);
//  seed=newSeed;
}

/**
Generates the next pseudorandom number. Subclasses should override this, as this is used by all other methods.

The general contract of next is that it returns an int value and if the argument bits is between 1 and 32 (inclusive), then that many low-order bits of the returned value will be (approximately) independently chosen bit values, each of which is (approximately) equally likely to be 0 or 1. The method next is implemented by class Random by atomically updating the seed to

 (seed * 0x5DEECE66DL + 0xBL) & ((1L << 48) - 1)

and returning

 (int)(seed >>> (48 - bits)).
*/

int next(int bits) {
 seed=(seed * 0x5DEECE66DLL + 0xBLL) & ((1LL << 48) - 1);
 return (int)(seed >> (48 - bits));
}

void main (void) {
	int i;
	for (i=0;i<50;i++) {
		printf ("%d ", next(31));
	}
}

File: Python/rng.py
import sys
def printf(format, *args):
    sys.stdout.write(format % args)

seed=0x5DEECE66D;
def setSeed(new_seed):
    global seed
    seed = (new_seed ^ 0x5DEECE66D) & ((1 << 48) - 1)

def next(bits):
    global seed 
    seed = (seed * 0x5DEECE66D + 0xB) & ((1 << 48) - 1)
    return (seed >> (48 - bits))


for i in range (0,50):
	printf ("%d ", next(31))

